src/utilities/viewer.cpp

```diff
@@ -606,19 +618,20 @@ class TestViewerWindow final : private D3D11Resources {
       }
       // Fill
       case 'F':
-        mFillMode = static_cast<FillMode>(
-          (static_cast<std::underlying_type_t<FillMode>>(mFillMode) + 1)
+        mSettings.mFillMode = static_cast<FillMode>(
+          (static_cast<std::underlying_type_t<FillMode>>(mSettings.mFillMode)
+           + 1)
           % FillModeCount);
         this->PaintNow();
         return;
       // Streamer
       case 'S':
-        mStreamerMode = !mStreamerMode;
-        if (mStreamerMode) {
-          mStreamerModePreviousFillMode = mFillMode;
-          mFillMode = FillMode::ColorKey;
-        } else if (mFillMode == FillMode::ColorKey) {
-          mFillMode = mStreamerModePreviousFillMode;
+        mSettings.mStreamerMode = !mSettings.mStreamerMode;
+        if (mSettings.mStreamerMode) {
+          mStreamerModePreviousFillMode = mSettings.mFillMode;
+          mSettings.mFillMode = FillMode::ColorKey;
+        } else if (mSettings.mFillMode == FillMode::ColorKey) {
+          mSettings.mFillMode = mStreamerModePreviousFillMode;

```

Comment:
This should also be handled in the settings saving/restoring.

Probably best to just also move previous fill mode to be in the settings too; an alternative would be to store/restore previous fill mode if streamer mode is enabled and the active fill mode otherwise.

Reply:
Not really sure what you want me to change here other than saving the previous fill mode in the settings too, which would make that setting user-editable. If they then mess around in the json file, they could potentially be messing up the cycle order when pressing F button. I've made `mStreamerModePreviousFillMode` be `Default` by default, so while in streamer mode and the application restarts, pressing F would result in the default filled background. ( it resulted in a black screen if the default was not set )

Reply:
Leaving for later, possibly removing mPreivousFillMode entirely and making it so streamer mode just ignores fill mode